app/v1/models/response/listPayments/ListPaymentsResponse.scala

```diff
@@ -16,13 +16,35 @@
 
 package v1.models.response.listPayments
 
+import cats.Functor
+import config.AppConfig
 import play.api.libs.json._
+import v1.hateoas.{HateoasLinks, HateoasListLinksFactory}
+import v1.models.hateoas.{HateoasData, Link}
 
-case class ListPaymentsResponse(payments: Seq[Payment])
+case class ListPaymentsResponse[I](payments: Seq[I])
```

Comment:
Is there any way we can implement this without making the list generic?

Reply:
We can but cannot use the existing wrapList function

Reply:
Ok, no worries. I just thought it might be a little overkill to have the response be generic when we know it must always return a sequence of payments.